substratevm/src/com.oracle.svm.agent/src/com/oracle/svm/agent/BreakpointInterceptor.java

```diff
@@ -132,7 +140,8 @@
     private static NativeImageAgent agent;
 
     private static Map<Long, Breakpoint> installedBreakpoints;
-
+    private static Set<String> definedClasses = new HashSet<>();
```

Comment:
I think this data structure needs to handle concurrent updates?

Reply:
fixed